app/controllers/coronavirus/announcements_controller.rb

```diff
@@ -10,27 +10,34 @@ def new
 
     def create
       @announcement = page.announcements.new(announcement_params)
-      if @announcement.save && draft_updater.send
-        redirect_to coronavirus_page_path(page.slug), notice: helpers.t("coronavirus.announcements.create.success")
-      else
-        render :new
+
+      unless @announcement.valid?
+        render :new, status: :unprocessable_entity
+        return
       end
+
+      Announcement.transaction do
+        @announcement.save!
+        draft_updater.send
+      end
+
+      redirect_to coronavirus_page_path(page.slug), notice: helpers.t("coronavirus.announcements.create.success")
+    rescue Pages::DraftUpdater::DraftUpdaterError => e
+      flash[:alert] = e.message
```

Comment:
I think this will need to be `flash.now` - I think you might find that these flash currently sticks around for the request after this one.